optional_plugins/golang/setup.py

```diff
@@ -13,10 +13,55 @@
 # Copyright: Red Hat Inc. 2017
 # Author: Amador Pahim <apahim@redhat.com>
 
-from setuptools import find_packages, setup
+import glob
+
+from setuptools import Command, find_packages, setup
 
 VERSION = open("VERSION", "r").read().strip()
 
+
+class Test(Command):
+    """Run tests"""
+
+    description = "Run tests"
+    user_options = []
+
+    def run(self):
+        # This must go there otherwise avocado must be installed for all targets
+        from avocado.core.job import Job
+        from avocado.core.suite import TestSuite
+        suites = []
+        pattern = 'tests/*'
+        config_check = {
+            'run.references': glob.glob(pattern),
+            'run.test_runner': 'nrunner',
+            'run.ignore_missing_references': True,
```

Comment:
Because this is a flat structure, I'd go with `run.references`  set to `glob.glob(tests/test_*py)` and remove `run.ignore_missing_references` because if we drop a broken `test_something.py`, we'd caught the error:

```
[cleber@p50 golang]$ dd bs=1k count=1 if=/dev/urandom of=tests/test_broken.py 
1+0 records in
1+0 records out
1024 bytes (1.0 kB, 1.0 KiB) copied, 0.000342099 s, 3.0 MB/s
[cleber@p50 golang]$ avocado run tests/test_*.py 
Could not resolve references: tests/test_broken.py
```

The other possibility is simply to rely on the extended file/directory feature of the resolver, and simply use `tests/` without any glob.

Reply:
ok, we'd have to change this in `check.py `as well